Add dedicated documentation for shared links (#840)
box/box-java-sdk: doc/files.md

> 

Reviewer: ```suggestion

A set of shared link access level constants are available through the SDK for convenience:
* `BoxSharedLink.Access.OPEN`
* `BoxSharedLink.Access.COLLABORATORS`
* `BoxSharedLink.Access.COMPANY`
* `BoxSharedLink.Access.DEFAULT`
* `accessLevels.DISABLED`

```

[create-shared-link]: http://opensource.box.com/box-java-sdk/javadoc/com/box/sdk/BoxFile.html#createSharedLink-com.box.sdk.BoxSharedLink.Access-java.util.Date-com.box.sdk.BoxSharedLink.Permissions-

Get a Shared Link
-----------------

> A shared link for a file can be generated by calling

Reviewer: ```suggestion
Retrieve the shared link for a file by calling
```
